src/components/utils/test/singleton_test.cc

```diff
@@ -192,14 +210,18 @@ TEST(SingletonTest, CreateDeleteSingletonInDifferentThreads) {
 }
 #endif
 
-TEST(SingletonTest, DISABLED_DeleteSingletonInDifferentThread) {
+TEST(SingletonTest, DeleteSingletonInDifferentThread) {
   // arrange
+  MainAndWorkerThreadSynchronisaiton job_sync;
   SingletonTest::instance();
   ASSERT_TRUE(SingletonTest::exists());
 
   threads::Thread* thread1 = threads::CreateThread(
-      "test thread", new DestroySingletonDelegate(SingletonTest::instance()));
+      "test thread",
+      new DestroySingletonDelegate(SingletonTest::instance(), job_sync));
+
   thread1->start();
+  job_sync.WaitJobDone(100);
```

Comment:
@LuxoftAKutsan, this method makes main UT thread to wait worker thread to finish the requested job. It must be called immediately after thread1->start().